script.harmony.control/resources/lib/kodisettings.py

```diff
@@ -0,0 +1,54 @@
+#v.0.3.4
+
+from kodi_six import xbmc, xbmcaddon
+
+ADDON         = xbmcaddon.Addon()
+ADDONNAME     = ADDON.getAddonInfo('id')
+ADDONLONGNAME = ADDON.getAddonInfo('name')
+ADDONVERSION  = ADDON.getAddonInfo('version')
+ADDONPATH     = ADDON.getAddonInfo('path')
+ADDONDATAPATH = xbmc.translatePath( ADDON.getAddonInfo('profile') )
+ADDONICON     = xbmc.translatePath('%s/icon.png' % ADDONPATH )
+ADDONLANGUAGE = ADDON.getLocalizedString
+SKINNAME      = xbmc.getSkinDir()
+
+
+def _get_setting( setting_name, default, thetype="string" ):
+    if thetype.lower() == "bool":
+        if ADDON.getSetting( setting_name ).lower() == 'true':
```

Comment:
You could use `ADDON.getSettingBool()` it was introduced in Kodi Leia. See https://codedocs.xyz/AlwinEsch/kodi/group__python__xbmcaddon.html.

Reply:
Historically I have had problems with getSettingBool consistently returning a boolean value.  I think it might have gotten fixed at some point, but periodically it returns an integer 0 instead. So I stuck with the standard getSetting and converted it on my own.